### PR TITLE
Completely refactor rmgpy.rmg.mainTest.TestMain

### PR DESCRIPTION
**Pull request by mliu49**
- Create permanent input file so it is not written every time
- Split tests into multiple test methods
- Add tests for edge seed libraries
- Clean up properly after tests

### CODE REVIEW

File: rmgpy/rmg/mainTest.py [hunk 1]
```diff
@@ -43,136 +43,143 @@
...
+
+        Note: Since this test modifies the class level RMG instance,
+        it can cause other tests to fail if run out of order."""
+        # Load the seed libraries into the database
+        self.rmg.database.load(
+            path=self.databaseDirectory,
+            thermoLibraries=['testSeed_edge'],
+            reactionLibraries=['testSeed_edge'],
+            seedMechanisms=['testSeed_edge'],
+            kineticsFamilies='none',
```
Comment (mjohnson541) [inline rmgpy/rmg/mainTest.py]: After the next seed mech pull we will try to load them as template reactions so we need to have the families loaded so can you load the default families?

Comment (goldmanm) [inline rmgpy/rmg/mainTest.py]: What do you mean by 'them'?

I don't understand why you need default families loaded. Which PR are you talking about? A specific number and link would be helpful.

Comment (mjohnson541) [inline rmgpy/rmg/mainTest.py]: Sorry, #1155 starts loading reactions as their original sources (template reactions and original library reactions).  If the family or library use to create a given reaction isn't loaded it can't find the original template/library and errors.  I mean I can change it later in that pull request of course, but I think it would be cleaner to do it here.

Comment (mliu49) [inline rmgpy/rmg/mainTest.py]: I understand what you mean. I removed them because it cut off 2 minutes from the test run time. I can put it back in.

**Comment by mliu49** [comment]
Updated and ready to merge. Kinetics families are now being loaded again.
